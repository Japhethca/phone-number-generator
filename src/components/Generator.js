import React, { useEffect, useState } from 'react';
import { css } from 'glamor';

import { withRedux } from '../hoc/withRedux';
import { getNumbersAction, generateNumbersAction } from '../store';
import PhoneNumbers from './PhoneNumbers';
import Controller from './Controller'


const generatorStyles = css({
  display: 'flex',
  flexDirection: 'column',
  width: '70%',
  margin: 'auto',
  backgroundColor: '#000000',
  color: 'green',
  minHeight: '60vh',
  borderRadius: '3px',
  padding: '20px'
});


export function Generator({ getNumbers, state, generateNumbers }) {
  const [noOfNumbers, setNoOfNumbers] = useState(0)
  useEffect(() => {
    if (state.numbers.length < 1 && !state.isFetchingNumbers) {
      getNumbers();
    }
  }, [state.isFetchingNumbers]);
  return (
    <div {...generatorStyles}>
      <Controller 
        setNoOfNumbers={setNoOfNumbers} 
        noOfNumbers={noOfNumbers}
        generateNumbers={generateNumbers}
        getNumbers={getNumbers}
      />
      <PhoneNumbers 
        numbers={state.numbers} 
        isFetchingNumbers={state.isFetchingNumbers}
      />
    </div>
  );
}

const dispatchActions = {
  getNumbers: getNumbersAction,
  generateNumbers: generateNumbersAction,
};

export default withRedux(Generator, dispatchActions);