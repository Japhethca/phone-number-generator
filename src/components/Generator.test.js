import React from 'react';
import renderer from 'react-test-renderer';

import { Generator } from './Generator';

describe('<Generator />', () => {
  const props = {
    getNumbers: jest.fn(), 
    state: {
      numbers: [],
      isFetchingNumbers: false,
    }, 
    generateNumbers: jest.fn(),
  };


  it('should render successfully', () => {
    const tree = renderer.create(<Generator {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
