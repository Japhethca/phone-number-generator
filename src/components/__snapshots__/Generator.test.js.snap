// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Generator /> should render successfully 1`] = `
<div
  data-css-l68izp=""
>
  <div
    data-css-1ugxpw4=""
  >
    <input
      data-css-1oss144=""
      maxLength="8"
      onChange={[Function]}
      placeholder="size of numbers"
      required={true}
      type="number"
      value={0}
    />
    <button
      data-css-fs9jv2=""
      id="button"
      onClick={[Function]}
    >
      Generate Numbers
    </button>
    <div>
      <select
        data-css-1i5es5r=""
        id=""
        name="sortOrder"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Sort Order
        </option>
        <option
          value="ASC"
        >
          Ascending
        </option>
        <option
          value="DESC"
        >
          Descending
        </option>
      </select>
    </div>
  </div>
  <div
    data-css-g7q9rr=""
  >
    <ol
      data-css-1j4x2ld=""
      id="copyNumbers"
    />
  </div>
</div>
`;
